src/components/projects/OrderedListToolbarItem.tsx

```diff
@@ -0,0 +1,251 @@
+  display: flex;
+  align-items: center;
+  gap: 6px;
+`
+
+const Block = styled.div`
+  height: 3px;
+  width: ${(props) => props.theme.grid.unit * 14}px;
+  background: ${(props) => props.theme.colors.border.tertiary};
+`
+
+const Label = styled.div`
+  font-family: Lato, serif;
+  font-size: ${(props) => props.theme.font.size.small};
+  font-weight: ${(props) => props.theme.font.weight.normal};
+  line-height: ${(props) => props.theme.font.lineHeight.small};
+  font-style: normal;
+`
+
+export const OrderedListToolbarItem: React.FC<{
+  state: EditorState
+  dispatch: (tr: Transaction) => void
+  view?: ManuscriptEditorView
+  config: ToolbarButtonConfig
+}> = ({ state, dispatch, view, config }) => {
+  const isEnabled = !config.isEnabled || config.isEnabled(state)
+
+  /**
+   *  When choosing a list type will do one of these options:
+   *  * if text is selected will wrap it with an order list
+   *  * if the selection is on list will change just the list type and
+   *    **in case it's nested list will change just the lists at the same level**
+   */
+  const handleClick = useCallback(
+    (style) => {
+      if (!dispatch) {
+        return
+      }
+
+      const type = style.type
+      const { $from, $to } = state.selection
+      const range = $from.blockRange($to)
+      const isListNode =
+        range &&
+        schema.nodes.ordered_list.compatibleContent(
+          $from.node(range.depth - 1).type
+        )
+
+      if (!isListNode) {
+        const options = config.options
+        if (options) {
+          options[type](state, dispatch, { listStyleType: type })
+          view && view.focus()
+        }
+      } else {
+        updateListStyle(state, dispatch, range, type)
+      }
+    },
+    [config, dispatch, state, view]
+  )
+
+  return (
+    <ToolbarItem>
+      <ListButton
+        title={config.title}
+        data-active={config.isActive && config.isActive(state)}
+        disabled={!isEnabled}
+        onMouseDown={(event) => {
+          event.preventDefault()
+          config.run(state, dispatch)
+          view && view.focus()
+        }}
+      >
+        <OrderedList />
+      </ListButton>
+      <ListStyleSelector
+        disabled={!isEnabled}
+        handleClick={handleClick}
+        styles={[
+          { items: ['1.', '2.', '3.'], type: 'order' },
+          { items: ['A.', 'B.', 'C.'], type: 'alpha-upper' },
+          { items: ['a.', 'b.', 'c.'], type: 'alpha-lower' },
+          { items: ['I.', 'II.', 'III.'], type: 'roman-upper' },
+          { items: ['i.', 'ii.', 'iii.'], type: 'roman-lower' },
+        ]}
+      />
+    </ToolbarItem>
+  )
+}
+
+/**
+ *  Will use node depth as an indicator for the list level, and `sharedDepth` will
+ *  give us all depths for the selection so we can find the parent list
+ *  After that will look at the nodes in the parent list and just update
+ *  the list that had the same level to the target level user has selected
+ */
+export const updateListStyle = (
```

Comment:
In my opinion all this functionality should be in the body-editor and just imported to be used in the tsx components here.

Reply:
Agreed. I've moved the logic to body-editor and simplified it as it was convoluted for what it does.